subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -1025,10 +1067,76 @@ static inline void aux_release(struct ll_adv_aux_set *aux)
 	mem_release(aux, &adv_aux_free);
 }
 
-inline uint8_t ull_adv_aux_handle_get(struct ll_adv_aux_set *aux)
+static uint16_t aux_time_get(struct ll_adv_aux_set *aux, struct pdu_adv *pdu,
+			     struct pdu_adv *pdu_scan)
 {
-	return mem_index_get(aux, ll_adv_aux_pool,
-			     sizeof(struct ll_adv_aux_set));
+	struct lll_adv_aux *lll_aux;
+	struct lll_adv *lll;
+	uint16_t time_us;
+
+	lll_aux = &aux->lll;
+	lll = lll_aux->adv;
+	time_us = PKT_AC_US(pdu->len, lll->phy_s) + EVENT_OVERHEAD_START_US +
+		  EVENT_OVERHEAD_END_US;
+
+	if (pdu->adv_ext_ind.adv_mode & BT_HCI_LE_ADV_PROP_CONN) {
+		const uint16_t conn_req_us =
+			PKT_AC_US((INITA_SIZE + ADVA_SIZE + LLDATA_SIZE),
+				  lll->phy_s);
+		const uint16_t conn_rsp_us =
+			PKT_AC_US((PDU_AC_EXT_HEADER_SIZE_MIN + ADVA_SIZE +
+				   TARGETA_SIZE), lll->phy_s);
+
+		time_us += EVENT_IFS_MAX_US * 2 + conn_req_us + conn_rsp_us;
+	} else if (pdu->adv_ext_ind.adv_mode & BT_HCI_LE_ADV_PROP_SCAN) {
```

Comment:
This may lead to error in if reserved value (0b11) is changed to non-reserved in BT Core spec. releases. More defense approach would be to use regular comparison instead of bitwise-and.